newdle/client/src/components/AnswerCalendar.js

```diff
@@ -0,0 +1,175 @@
+      action: addAnswer(slot, 'ifneedbe'),
+      style: styles['available'],
+    };
+  } else if (answer === 'ifneedbe') {
+    return {
+      icon: 'minus square outline',
+      action: removeAnswer(slot),
+      style: styles['ifneedbe'],
+    };
+  } else {
+    return {
+      icon: 'square outline',
+      action: addAnswer(slot, 'available'),
+      style: styles['unavailable'],
+    };
+  }
+}
+
+function getSlotProps(slot, duration, minHour, maxHour, answer) {
+  const start = toMoment(slot, moment.HTML5_FMT.DATETIME_LOCAL);
+  const end = toMoment(start).add(duration, 'm');
+  const answerProps = getAnswerProps(slot, answer);
+
+  return {
+    slot,
+    startTime: serializeDate(start, 'HH:mm'),
+    endTime: serializeDate(end, 'HH:mm'),
+    height: calculateHeight(start, end, minHour, maxHour),
+    pos: calculatePosition(start, minHour, maxHour),
+    key: slot,
+    answer: answer || 'unavailable',
+    ...answerProps,
+  };
+}
+
+function calculateOptionsPositions(options, duration, minHour, maxHour, answers) {
+  const optionsByDate = _.groupBy(
+    options.map(slot => getSlotProps(slot, duration, minHour, maxHour, answers[slot])),
+    slot => serializeDate(toMoment(slot.slot, moment.HTML5_FMT.DATETIME_LOCAL))
+  );
+
+  return Object.entries(optionsByDate).map(([date, options]) => {
+    return {date, optionGroups: groupOverlaps(options)};
+  });
+}
+
+function Hours({minHour, maxHour, hourStep}) {
+  const hourSeries = _.range(minHour, maxHour + hourStep, hourStep);
+  const hourSpan = maxHour - minHour;
+
+  return (
+    <div className={styles['hours-column']}>
+      {_.range(0, hourSpan + hourStep, hourStep).map((i, n) => (
+        <div
+          className={styles['hour']}
+          key={`hour-label-${i}`}
+          style={{top: `${(i / hourSpan) * 100}%`}}
+        >
+          {serializeDate(toMoment({hours: hourSeries[n]}), 'k')}
+        </div>
+      ))}
+    </div>
+  );
+}
+
+Hours.propTypes = {
+  minHour: PropTypes.number.isRequired,
+  maxHour: PropTypes.number.isRequired,
+  hourStep: PropTypes.number,
+};
+
+Hours.defaultProps = {
+  hourStep: 2,
+};
+
+export default function AnswerCalendar({minHour, maxHour}) {
+  const answers = useSelector(getAnswers);
+  const timeSlots = useSelector(getNewdleTimeslots);
+  const duration = useSelector(getNewdleDuration);
+  const optionsByDay = calculateOptionsPositions(timeSlots, duration, minHour, maxHour, answers);
+
+  if (optionsByDay.length === 0) {
+    return <div>No data</div>;
+  }
+
+  return (
+    <Grid>
+      <Grid.Row>
+        <Grid.Column width={1}>
+          <Hours minHour={minHour} maxHour={maxHour} />
+        </Grid.Column>
+        {optionsByDay.slice(0, 3).map(options => (
+          <Grid.Column width={5}>
+            <AnswerCalendarDay options={options} key={options.date} />
+          </Grid.Column>
+        ))}
+      </Grid.Row>
+    </Grid>
+  );
+}
+
+AnswerCalendar.propTypes = {
+  minHour: PropTypes.number,
```

Comment:
I think this component needs the same shift/expand logic we added on the creation page if the default times don't cover everything. i.e. if shift the min/max hours to cover everything, or if that's not enough extend it to a longer range